data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java

```diff
@@ -276,6 +276,20 @@ public String toJsonString() {
         return jsonNode.toString();
     }
 
+    @Override
+    public String toJsonStringWithTags(final String tagsKey) {
+        String result = jsonNode.toString();
+        String separator = "";
+
+        result = result.substring(0, result.length()-1) + ",\"tags\":[";
+        for (final String tag: getMetadata().getTags()) {
```

Comment:
I don't think we should be manipulating JSON manually. Let's do this with Jackson.

You can create a `JsonNode` and then add the value.